### PR TITLE
dns reporting 

### PR DESCRIPTION
**Pull request by 0xmchadha**
We will be reporting dns lookups being done by the PU's. This PR adds the gaia spec and the corresponding generated go file.

### CODE REVIEW

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-proxy",
    "component": "gaia",
    "pr-id": "388",
    "commit-sha": "ce7f3becf8f13f5d479bf588cba5b55397d850c1"
  },
  {
    "project": "dns-proxy",
    "component": "trireme-lib",
    "pr-id": "854",
    "commit-sha": "886e16f92fd6c91f67b73adcdeefc816a247d7a7"
  },
  {
    "project": "dns-proxy",
    "component": "enforcerd",
    "pr-id": "1325",
    "commit-sha": "e7badd413650fafe910a538aac28707bd2cdcdbe"
  }
]
```

File: specs/dnsreport.spec [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Model
+model:
+  rest_name: dnsreport
+  resource_name: dnsreports
+  entity_name: DNSReport
+  package: zack
+  group: policy/networking
+  description: Post a new dns request report.
```
Comment (t00f) [inline specs/dnsreport.spec]: Descriptions will appear in the API documentation. Can you add more details please ? What is the source of the report for instance?

File: specs/dnsreport.spec [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+# Attributes
+attributes:
+  v1:
+  - name: sourceNamespace
+    description: Namespace of the source.
+    type: string
+    exposed: true
+    example_value: /my/namespace
```
Comment (t00f) [inline specs/dnsreport.spec]: All report needs a `namespace`. I only see a `sourceNamespace`. What does it represent ? What is the source here. Is it the enforcer ?

File: specs/dnsreport.spec [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+  - name: sourceID
+    description: ID of the source.
+    type: string
+    exposed: true
+    required: true
+    example_value: xxx-xxx-xxx
```
Comment (t00f) [inline specs/dnsreport.spec]: Same here, what is the source ? Is it a PU ?

File: specs/dnsreport.spec [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    exposed: true
+    required: true
+    example_value: www.google.com
+
+  - name: result
+    description: Result reports whether dns request succeeded or failed.
+    type: boolean
+    exposed: true
+    required: true
+    example_value: true
```
Comment (t00f) [inline specs/dnsreport.spec]: Usually, we call this property `success`. Check out the `auditreport.spec`:

```
  - name: success
    description: Tells if the operation has been a success or a failure.
    type: boolean
    exposed: true
```

File: specs/dnsreport.spec [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+  - name: error
```
Comment (t00f) [inline specs/dnsreport.spec]: Usually, this is called the reason:
```
  - name: reason
    description: |-
      This field is only set if `action` is set to `Reject`. It specifies the reason
      for the rejection.
    type: string
    exposed: true
```

File: specs/dnsreport.spec [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    description: If the result is false, error reports the reason of the dns failure.
+    type: string
+    exposed: true
+
+  - name: count
+    description: Number of times the client saw this activity.
+    type: integer
+    exposed: true
+    required: true
+    example_value: 1
```
Comment (t00f) [inline specs/dnsreport.spec]: Influx db always need a `value`. Here is what we use usually:
```
  - name: value
    description: Number of flows in the log.
    type: integer
    exposed: true
    required: true
    example_value: 1
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-proxy",
    "component": "gaia",
    "pr-id": "388",
    "commit-sha": "ce7f3becf8f13f5d479bf588cba5b55397d850c1"
  },
  {
    "project": "dns-proxy",
    "component": "trireme-lib",
    "pr-id": "854",
    "commit-sha": "3f2dc70f68d3e1b40a07a3f048ed1245eba40442"
  },
  {
    "project": "dns-proxy",
    "component": "enforcerd",
    "pr-id": "1325",
    "commit-sha": "e7badd413650fafe910a538aac28707bd2cdcdbe"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-proxy",
    "component": "gaia",
    "pr-id": "388",
    "commit-sha": "ce7f3becf8f13f5d479bf588cba5b55397d850c1"
  },
  {
    "project": "dns-proxy",
    "component": "trireme-lib",
    "pr-id": "854",
    "commit-sha": "3cc0419b5d13b3037fc2121ef9617b2486c89d64"
  },
  {
    "project": "dns-proxy",
    "component": "enforcerd",
    "pr-id": "1325",
    "commit-sha": "8d3fdc26042beb68af738b486732bc334e20f0bb"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-proxy",
    "component": "trireme-lib",
    "pr-id": "854",
    "commit-sha": "2a7636a360f8feecf90f2a700729ae5a19f65e49"
  },
  {
    "project": "dns-proxy",
    "component": "enforcerd",
    "pr-id": "1325",
    "commit-sha": "8d3fdc26042beb68af738b486732bc334e20f0bb"
  },
  {
    "project": "dns-proxy",
    "component": "gaia",
    "pr-id": "388",
    "commit-sha": "2dfd8140b34ee738e430ceba31d0e70b37f90db7"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-proxy",
    "component": "trireme-lib",
    "pr-id": "854",
    "commit-sha": "2329f768a274586a2f00074742cb32d9a73275de"
  },
  {
    "project": "dns-proxy",
    "component": "enforcerd",
    "pr-id": "1325",
    "commit-sha": "8aab6d1a53f73bfacf9304806480874db6450a24"
  },
  {
    "project": "dns-proxy",
    "component": "apotests",
    "pr-id": "2229",
    "commit-sha": "0a8330240880d6f41b4569a9ce31039438a0003d"
  },
  {
    "project": "dns-proxy",
    "component": "backend",
    "pr-id": "524",
    "commit-sha": "58ca346b7ab2a45fce740f5ecdddd7248532e961"
  },
  {
    "project": "dns-proxy",
    "component": "apoctl",
    "pr-id": "197",
    "commit-sha": "548e99cbe02aad59648bf6bc134805b19bb1cd0c"
  },
  {
    "project": "dns-proxy",
    "component": "gaia",
    "pr-id": "388",
    "commit-sha": "ce1e4d25185549659d1b2fe8666d6249ddb24161"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-proxy",
    "component": "apotests",
    "pr-id": "2229",
    "commit-sha": "0a8330240880d6f41b4569a9ce31039438a0003d"
  },
  {
    "project": "dns-proxy",
    "component": "backend",
    "pr-id": "524",
    "commit-sha": "19b2800a1849baee4e2cf2333c0d7f06d5932557"
  },
  {
    "project": "dns-proxy",
    "component": "apoctl",
    "pr-id": "197",
    "commit-sha": "548e99cbe02aad59648bf6bc134805b19bb1cd0c"
  },
  {
    "project": "dns-proxy",
    "component": "gaia",
    "pr-id": "388",
    "commit-sha": "ce1e4d25185549659d1b2fe8666d6249ddb24161"
  },
  {
    "project": "dns-proxy",
    "component": "trireme-lib",
    "pr-id": "854",
    "commit-sha": "2329f768a274586a2f00074742cb32d9a73275de"
  },
  {
    "project": "dns-proxy",
    "component": "enforcerd",
    "pr-id": "1325",
    "commit-sha": "8aab6d1a53f73bfacf9304806480874db6450a24"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-proxy",
    "component": "apotests",
    "pr-id": "2229",
    "commit-sha": "0a8330240880d6f41b4569a9ce31039438a0003d"
  },
  {
    "project": "dns-proxy",
    "component": "backend",
    "pr-id": "524",
    "commit-sha": "19b2800a1849baee4e2cf2333c0d7f06d5932557"
  },
  {
    "project": "dns-proxy",
    "component": "apoctl",
    "pr-id": "197",
    "commit-sha": "548e99cbe02aad59648bf6bc134805b19bb1cd0c"
  },
  {
    "project": "dns-proxy",
    "component": "gaia",
    "pr-id": "388",
    "commit-sha": "ce1e4d25185549659d1b2fe8666d6249ddb24161"
  },
  {
    "project": "dns-proxy",
    "component": "trireme-lib",
    "pr-id": "854",
    "commit-sha": "2329f768a274586a2f00074742cb32d9a73275de"
  },
  {
    "project": "dns-proxy",
    "component": "enforcerd",
    "pr-id": "1325",
    "commit-sha": "edab1ab10775ef9e300bec8a28c6d273a29d2a78"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "gaia",
    "pr-id": "388",
    "commit-sha": "c8478f03e1c28ce41d5b7c7a8d2ef7f0c1d46018"
  },
  {
    "project": "",
    "component": "apoctl",
    "pr-id": "197",
    "commit-sha": "5378cfb89c69dd73775921f929a82538d00bbdea"
  },
  {
    "project": "",
    "component": "backend",
    "pr-id": "524",
    "commit-sha": "347460feb01c9f508ac2fc11423ceff19d7b8cf0"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-proxy",
    "component": "backend",
    "pr-id": "524",
    "commit-sha": "347460feb01c9f508ac2fc11423ceff19d7b8cf0"
  },
  {
    "project": "dns-proxy",
    "component": "apoctl",
    "pr-id": "197",
    "commit-sha": "5378cfb89c69dd73775921f929a82538d00bbdea"
  },
  {
    "project": "dns-proxy",
    "component": "gaia",
    "pr-id": "388",
    "commit-sha": "c8478f03e1c28ce41d5b7c7a8d2ef7f0c1d46018"
  },
  {
    "project": "dns-proxy",
    "component": "trireme-lib",
    "pr-id": "854",
    "commit-sha": "2329f768a274586a2f00074742cb32d9a73275de"
  },
  {
    "project": "dns-proxy",
    "component": "enforcerd",
    "pr-id": "1325",
    "commit-sha": "93964281d2e7be5ff97efe1267e6194c96938f82"
  },
  {
    "project": "dns-proxy",
    "component": "apotests",
    "pr-id": "2229",
    "commit-sha": "0a8330240880d6f41b4569a9ce31039438a0003d"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-proxy",
    "component": "apotests",
    "pr-id": "2229",
    "commit-sha": "0a8330240880d6f41b4569a9ce31039438a0003d"
  },
  {
    "project": "dns-proxy",
    "component": "backend",
    "pr-id": "524",
    "commit-sha": "347460feb01c9f508ac2fc11423ceff19d7b8cf0"
  },
  {
    "project": "dns-proxy",
    "component": "apoctl",
    "pr-id": "197",
    "commit-sha": "5378cfb89c69dd73775921f929a82538d00bbdea"
  },
  {
    "project": "dns-proxy",
    "component": "gaia",
    "pr-id": "388",
    "commit-sha": "c8478f03e1c28ce41d5b7c7a8d2ef7f0c1d46018"
  },
  {
    "project": "dns-proxy",
    "component": "trireme-lib",
    "pr-id": "854",
    "commit-sha": "2329f768a274586a2f00074742cb32d9a73275de"
  },
  {
    "project": "dns-proxy",
    "component": "enforcerd",
    "pr-id": "1325",
    "commit-sha": "29a077d5be4bad1d31a67084c59eeabd33d25e5f"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "backend",
    "pr-id": "524",
    "commit-sha": "347460feb01c9f508ac2fc11423ceff19d7b8cf0"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "388",
    "commit-sha": "2c93822e9e29c36ff0d41b5ba9e195dbc4ec843e"
  },
  {
    "project": "",
    "component": "apoctl",
    "pr-id": "197",
    "commit-sha": "5378cfb89c69dd73775921f929a82538d00bbdea"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "apoctl",
    "pr-id": "197",
    "commit-sha": "5378cfb89c69dd73775921f929a82538d00bbdea"
  },
  {
    "project": "",
    "component": "backend",
    "pr-id": "524",
    "commit-sha": "347460feb01c9f508ac2fc11423ceff19d7b8cf0"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "388",
    "commit-sha": "34a0889300d27c5d642c88ebf9b95bf662d89b09"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-proxy",
    "component": "backend",
    "pr-id": "524",
    "commit-sha": "347460feb01c9f508ac2fc11423ceff19d7b8cf0"
  },
  {
    "project": "dns-proxy",
    "component": "apoctl",
    "pr-id": "197",
    "commit-sha": "5378cfb89c69dd73775921f929a82538d00bbdea"
  },
  {
    "project": "dns-proxy",
    "component": "gaia",
    "pr-id": "388",
    "commit-sha": "34a0889300d27c5d642c88ebf9b95bf662d89b09"
  },
  {
    "project": "dns-proxy",
    "component": "trireme-lib",
    "pr-id": "854",
    "commit-sha": "d82a148d32c81389b9ed16ffa5acd11e271bbfb1"
  },
  {
    "project": "dns-proxy",
    "component": "enforcerd",
    "pr-id": "1325",
    "commit-sha": "9c502ed27ab4a4edaf05a9fad72169a013175668"
  },
  {
    "project": "dns-proxy",
    "component": "apotests",
    "pr-id": "2229",
    "commit-sha": "0a8330240880d6f41b4569a9ce31039438a0003d"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-proxy",
    "component": "apoctl",
    "pr-id": "197",
    "commit-sha": "5378cfb89c69dd73775921f929a82538d00bbdea"
  },
  {
    "project": "dns-proxy",
    "component": "gaia",
    "pr-id": "388",
    "commit-sha": "0de87689e348fe48634e9f61d3346a5718157a95"
  },
  {
    "project": "dns-proxy",
    "component": "trireme-lib",
    "pr-id": "854",
    "commit-sha": "d4f35355fa234297f7e758a6c9fdf6f4a75b5f4e"
  },
  {
    "project": "dns-proxy",
    "component": "enforcerd",
    "pr-id": "1325",
    "commit-sha": "9c502ed27ab4a4edaf05a9fad72169a013175668"
  },
  {
    "project": "dns-proxy",
    "component": "apotests",
    "pr-id": "2229",
    "commit-sha": "0a8330240880d6f41b4569a9ce31039438a0003d"
  },
  {
    "project": "dns-proxy",
    "component": "backend",
    "pr-id": "524",
    "commit-sha": "347460feb01c9f508ac2fc11423ceff19d7b8cf0"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-proxy",
    "component": "trireme-lib",
    "pr-id": "854",
    "commit-sha": "901e6583bfa2242a8c066badd942cc6c06b380c1"
  },
  {
    "project": "dns-proxy",
    "component": "enforcerd",
    "pr-id": "1325",
    "commit-sha": "bf3dfd98fbc87335470104fcca111e4d4e6bb618"
  },
  {
    "project": "dns-proxy",
    "component": "apotests",
    "pr-id": "2229",
    "commit-sha": "c40d22cd1cd1eda77e0044588e93b15716f2b778"
  },
  {
    "project": "dns-proxy",
    "component": "backend",
    "pr-id": "524",
    "commit-sha": "347460feb01c9f508ac2fc11423ceff19d7b8cf0"
  },
  {
    "project": "dns-proxy",
    "component": "apoctl",
    "pr-id": "197",
    "commit-sha": "5378cfb89c69dd73775921f929a82538d00bbdea"
  },
  {
    "project": "dns-proxy",
    "component": "gaia",
    "pr-id": "388",
    "commit-sha": "0de87689e348fe48634e9f61d3346a5718157a95"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "backend",
    "pr-id": "524",
    "commit-sha": "347460feb01c9f508ac2fc11423ceff19d7b8cf0"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "388",
    "commit-sha": "0de87689e348fe48634e9f61d3346a5718157a95"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-proxy",
    "component": "apoctl",
    "pr-id": "197",
    "commit-sha": "5378cfb89c69dd73775921f929a82538d00bbdea"
  },
  {
    "project": "dns-proxy",
    "component": "gaia",
    "pr-id": "388",
    "commit-sha": "0de87689e348fe48634e9f61d3346a5718157a95"
  },
  {
    "project": "dns-proxy",
    "component": "trireme-lib",
    "pr-id": "854",
    "commit-sha": "901e6583bfa2242a8c066badd942cc6c06b380c1"
  },
  {
    "project": "dns-proxy",
    "component": "enforcerd",
    "pr-id": "1325",
    "commit-sha": "bf3dfd98fbc87335470104fcca111e4d4e6bb618"
  },
  {
    "project": "dns-proxy",
    "component": "apotests",
    "pr-id": "2229",
    "commit-sha": "c40d22cd1cd1eda77e0044588e93b15716f2b778"
  },
  {
    "project": "dns-proxy",
    "component": "backend",
    "pr-id": "524",
    "commit-sha": "347460feb01c9f508ac2fc11423ceff19d7b8cf0"
  }
]
```

**Pull request merged**